Sources/JavaScriptKit/FundamentalObjects/JSClosure.swift

```diff
@@ -0,0 +1,140 @@
+import _CJavaScriptKit
+
+fileprivate var sharedFunctions: [JavaScriptHostFuncRef: ([JSValue]) -> JSValue] = [:]
+
+/// `JSOneshotClosure` is a JavaScript function that can be called only once.
+public class JSOneshotClosure: JSFunction {
+    private var hostFuncRef: JavaScriptHostFuncRef = 0
+
+    public init(_ body: @escaping ([JSValue]) -> JSValue) {
+        // 1. Fill `id` as zero at first to access `self` to get `ObjectIdentifier`.
+        super.init(id: 0)
+        let objectId = ObjectIdentifier(self)
+        let funcRef = JavaScriptHostFuncRef(bitPattern: Int32(objectId.hashValue))
+        // 2. Retain the given body in static storage by `funcRef`.
+        sharedFunctions[funcRef] = body
+        // 3. Create a new JavaScript function which calls the given Swift function.
+        var objectRef: JavaScriptObjectRef = 0
+        _create_function(funcRef, &objectRef)
+
+        hostFuncRef = funcRef
+        id = objectRef
+    }
+
+    public override func callAsFunction(this: JSObject? = nil, arguments: [ConvertibleToJSValue]) -> JSValue {
+        defer { release() }
+        return super.callAsFunction(this: this, arguments: arguments)
+    }
```

Comment:
I’m wondering — if the function will be destroyed as soon as it’s called, does it even make sense to call a one-shot closure from Swift?

Reply:
Yes, usually users don't call `JSOneshotClosure` and `JSClosure` directly from Swift.
It might make sense to remove `JSFunction` inheritance and remove `callAsFunction`.